### PR TITLE
add track exception

### PR DESCRIPTION
**Pull request by markwolff**
- Add trackException to ApplicationInsightsAnalytics API

- Add option to ApplicationInsightsAnalytics config to enable auto collection of uncaught exceptions (this is opt-out, as it was in old SDK)

### CODE REVIEW

File: ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts [hunk 1]
```diff
@@ -199,7 +201,79 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
...
+                { exception: Util.dump(e) });
+        }
+    }
+
+    /**
+     * @description Custom error handler for Application Insights Analytics
+     * @param {IAutoExceptionTelemetry} exception
+     * @memberof ApplicationInsights
+     */
+    public _onerror(exception: IAutoExceptionTelemetry): void {
```
Comment (jpiyali) [inline ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts]: did this exist before?

File: AISKU/snippet/snippet.js [hunk 2]
```diff
@@ -31,14 +31,39 @@
...
 
-    var method = ["PageView"];
+    var method = ["PageView", "Exception"];
     while (method.length) {
         createLazyMethod("track" + method.pop());
     }
 
+    // Collect global errors
+    if (aiConfig.extensionsConfig &&
+        aiConfig.extensionsConfig.ApplicationInsightsAnalytics &&
```
Comment (jpiyali) [inline AISKU/snippet/snippet.js]: use aiConfig.extensionsConfig[this.identifier] as configurations are by plugin identifier

Comment (jpiyali) [reply AISKU/snippet/snippet.js]: or add comment here

File: ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts (hunk 1 above)
Comment (barustum) [reply ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts]: Yes. This is the method that will be called by the snippet (snippet.js line 53) in the case of auto collection. It essentially calls trackException. 

---
In reply to: [217542252](https://github.com/Microsoft/ApplicationInsights-JS/pull/668#discussion_r217542252) [](ancestors = 217542252)

File: AISKU/snippet/snippet.js (hunk 2 above)
Comment (markwolff) [reply AISKU/snippet/snippet.js]: I've added a comment to clarify 7c915f3

**Pull request merged**